trino-aws-proxy/src/main/java/io/trino/aws/proxy/plugins/filebasedcredentialsprovider/FileBasedCredentialsProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.plugins.filebasedcredentialsprovider;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import io.airlift.json.ObjectMapperProvider;
+import io.trino.aws.proxy.spi.credentials.Credentials;
+import io.trino.aws.proxy.spi.credentials.CredentialsProvider;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.Objects.requireNonNull;
+
+public class FileBasedCredentialsProvider
+        implements CredentialsProvider
+{
+    private final Map<String, Credentials> credentialsStore;
+
+    @Inject
+    public FileBasedCredentialsProvider(FileBasedCredentialsProviderConfig config, ObjectMapperProvider objectMapperProvider)
```

Comment:
Please inject an `ObjectMapper`, not `ObjectMapperProvider`

Reply:
Does airlift register the JDKmodule while injecting the ObjectMapper ?

Reply:
Yes